Add sonar analysis and badges to README and update .travis.yml to accomodate JDK8

lewismc opened the pull request:
Hi @jnioche and folks... this PR defines some nice badges for us to have insight into how code analysis and test coverage etc looks in CC.
It also updates the .travis.yml with the JDK8 config.

jnioche commented:
@lewismc shall we drop this for now and release 0.7?

lewismc commented:
Hi @jnioche apologies I dropped off of this one.
I've updated all merge conflicts... wee what you think now. BTW Sonar Analysis can be seen at https://sonarqube.com/dashboard/index/com.github.crawler-commons:crawler-commons

lewismc commented:
You'll also see that codecov is now integrated as well https://codecov.io/gh/crawler-commons/crawler-commons, once this is pushed we will now have automatic updates to tickets regarding code coverage stats.

jnioche commented:
Thanks @lewismc

[https://codecov.io/gh/crawler-commons/crawler-commons] probably requires some credentials, I am getting "Sorry, you do not have permission to the repository upload token"

[https://img.shields.io/github/license/crawler-commons/crawler-commons.svg?maxAge=2592000?style=plastic] => unknown license?

Can you keep the same formatting for the pom.xml? Having the whole file marked as different makes it impossible to review.

I can see at least one addition which you commented out  \: [https://github.com/crawler-commons/crawler-commons/pull/127/files#diff-600376dffeb79835ede4a0b285078036R182]. The POM file is crucial, let's keep it as simple and clean as possible.

All this increases the reliance on external services. For instance, issues can be opened in sonarqube. I am not super comfortable with putting stuff all over the place, but having said that what SonarQube produces is at least worth looking at. 

@kkrugler what do you think?

lewismc commented:
Hi julien thanks for review. I agreebout Pom.xml.
I'll address as much as I can, and also feedback from ken would also be
good.


On Wednesday, October 19, 2016, Julien Nioche <notifications@github.com>
wrote:

> Thanks @lewismc <https://github.com/lewismc>
>
> [https://codecov.io/gh/crawler-commons/crawler-commons] probably requires
> some credentials, I am getting "Sorry, you do not have permission to the
> repository upload token"
>
> [https://img.shields.io/github/license/crawler-
> commons/crawler-commons.svg?maxAge=2592000?style=plastic] => unknown
> license?
>
> Can you keep the same formatting for the pom.xml? Having the whole file
> marked as different makes it impossible to review.
>
> I can see at least one addition which you commented out : [
> https://github.com/crawler-commons/crawler-commons/pull/127/files#diff-
> 600376dffeb79835ede4a0b285078036R182]. The POM file is crucial, let's
> keep it as simple and clean as possible.
>
> All this increases the reliance on external services. For instance, issues
> can be opened in sonarqube. I am not super comfortable with putting stuff
> all over the place, but having said that what SonarQube produces is at
> least worth looking at.
>
> @kkrugler <https://github.com/kkrugler> what do you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/crawler-commons/crawler-commons/pull/127#issuecomment-254753630>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABHJl-ckQlaA4n96nzm-VCPLv_c0SSYQks5q1dqNgaJpZM4KHEZc>
> .
>


-- 

*Lewis*
Dr. Lewis J. McGibbney Ph.D, B.Sc
Director
Phone: +1(626)498-3090
Skype: lewis.john.mcgibbney
Email: lewis.mcgibbney@gmail.com

lewismc commented:
> [https://codecov.io/gh/crawler-commons/crawler-commons] probably requires some credentials, I am getting "Sorry, you do not have permission to the repository upload token"

I think this is because we have not officially pushed anything there yet.

> [https://img.shields.io/github/license/crawler-commons/crawler-commons.svg?maxAge=2592000?style=plastic] => unknown license?

This is because we do not have a LICENSE file in the codebase. I've updated the PR with one.

> Can you keep the same formatting for the pom.xml? Having the whole file marked as different makes it impossible to review.

Bugger, apologies I was not even aware that the formatting had changed. I've changed this back and the additions should be clear now @jnioche 

> I can see at least one addition which you commented out : [https://github.com/crawler-commons/crawler-commons/pull/127/files#diff-600376dffeb79835ede4a0b285078036R182]. The POM file is crucial, let's keep it as simple and clean as possible.

Again this was not intended. The aim is to create a new profile which is activated within the CI and NOT for default builds. This has been rectified as well.

> All this increases the reliance on external services. For instance, issues can be opened in sonarqube. I am not super comfortable with putting stuff all over the place, but having said that what SonarQube produces is at least worth looking at. 

Point taken. Personally I've had very good experiences with SonarQube both at the Apache-hosted server and via the online service which I am proposing for CC.

lewismc commented:
@kkrugler @jnioche did anyone get a chance to take a look? I am not fussed if this does not make it into the next release. If we need to hold off then that is fine by me.

jnioche commented:
thanks @lewismc 

+1 to the license and badge for it (could be done right now as a separate commit?)

I've found that sonar can be installed as a plugin for the main IDEs [http://www.sonarlint.org/]; some of the stuff it reports is useful but a lot of it is pretty inane. Having that as a service would mean having the same alerts about problems that aren't really important coming over and over.  We can always use the IDE plugins individually and commit changes as necessary + track all this in GitHub.

@kkrugler what do you think?

This should definitely not block a release of 0.7 and can be added any time later.

lewismc commented:
Hi @jnioche OK so

> +1 to the license and badge for it (could be done right now as a separate commit?)

done

> I've found that sonar can be installed as a plugin for the main IDEs [http://www.sonarlint.org/]; some of the stuff it reports is useful but a lot of it is pretty inane. Having that as a service would mean having the same alerts about problems that aren't really important coming over and over. We can always use the IDE plugins individually and commit changes as necessary + track all this in GitHub.

Fair points, it all depends upon preferences I think.

> This should definitely not block a release of 0.7 and can be added any time later. 

Agreed @jnioche if you want I can push a release candidate tonight.

jnioche commented:
> done

Thanks @lewismc 

> Agreed @jnioche if you want I can push a release candidate tonight.

that would be fab, thanks Lewis

kkrugler commented:
Regarding sonar - yes, lots of false positives, though some useful input. If there's a way to flag things as ignore (e.g. https://sonarqube.com/issues/search#issues=AVfY9C0TPl_GJI-SBVbO) then that would make it a lot more useful to me.

jnioche commented:
Close for now?

lewismc commented:
Yes @jnioche @kkrugler if we want, we can come back to this in the near future.
